frmts/esric/esric_dataset.cpp

```diff
@@ -352,6 +362,58 @@ CPLErr ECDataset::Initialize(CPLXMLNode *CacheInfo)
     return error;
 }
 
+static std::unique_ptr<OGRSpatialReference>
+CreateSRS(const CPLJSONObject &oSRSRoot)
+{
+    auto poSRS = std::make_unique<OGRSpatialReference>();
+
+    bool bSuccess = false;
+    const int nCode = oSRSRoot.GetInteger("wkid");
+    // The concept of LatestWKID is explained in
+    // https://support.esri.com/en/technical-article/000013950
+    const int nLatestCode = oSRSRoot.GetInteger("latestWkid");
+
+    // Try first with nLatestWKID as there is a higher chance it is a
+    // EPSG code and not an ESRI one.
+    if (nLatestCode > 0)
+    {
+        if (nLatestCode > 32767)
+        {
+            if (poSRS->SetFromUserInput(CPLSPrintf("ESRI:%d", nLatestCode)) ==
+                OGRERR_NONE)
+            {
+                bSuccess = true;
+            }
+        }
+        else if (poSRS->importFromEPSG(nLatestCode) == OGRERR_NONE)
+        {
+            bSuccess = true;
+        }
+    }
+    if (!bSuccess && nCode > 0)
+    {
+        if (nCode > 32767)
+        {
+            if (poSRS->SetFromUserInput(CPLSPrintf("ESRI:%d", nCode)) ==
+                OGRERR_NONE)
+            {
+                bSuccess = true;
+            }
+        }
+        else if (poSRS->importFromEPSG(nCode) == OGRERR_NONE)
+        {
+            bSuccess = true;
```

Comment:
ok, probably nitpicking but since it crossed my mind I'll tell: does this `return` from a branch messes with RVO ?

Reply:
Perhaps ? I don't think it matters much as this is not a performance critical code path. And as we return à unique_ptr I don't think any heavy object copying could be done as it can only be moved